alacritty_terminal/src/term/mod.rs

```diff
@@ -607,37 +612,76 @@ impl From<&BellConfig> for VisualBell {
 #[derive(Serialize, Deserialize, Debug, Copy, Clone, PartialEq)]
 pub struct SizeInfo {
     /// Terminal window width.
-    pub width: f32,
+    width: f32,
 
     /// Terminal window height.
-    pub height: f32,
+    height: f32,
 
     /// Width of individual cell.
-    pub cell_width: f32,
+    cell_width: f32,
 
     /// Height of individual cell.
-    pub cell_height: f32,
+    cell_height: f32,
 
     /// Horizontal window padding.
-    pub padding_x: f32,
+    padding_x: f32,
 
     /// Horizontal window padding.
-    pub padding_y: f32,
+    padding_y: f32,
+
+    /// Number of lines in the viewport.
+    screen_lines: Line,
+
+    /// Number of columns in the viewport.
+    cols: Column,
 
     /// DPR of the current window.
     #[serde(default)]
     pub dpr: f64,
 }
 
 impl SizeInfo {
-    #[inline]
-    pub fn lines(&self) -> Line {
-        Line(((self.height - 2. * self.padding_y) / self.cell_height) as usize)
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        width: f32,
+        height: f32,
+        cell_width: f32,
+        cell_height: f32,
+        padding_x: f32,
+        padding_y: f32,
+        dpr: f64,
+        dynamic_padding: bool,
+    ) -> SizeInfo {
+        let lines = (height - 2. * padding_y) / cell_height;
+        let screen_lines = Line(max(lines as usize, MIN_SCREEN_LINES));
+
+        let cols = (width - 2. * padding_x) / cell_width;
+        let cols = Column(max(cols as usize, MIN_COLS));
+
+        let mut padding_x = padding_x * dpr as f32;
+        let mut padding_y = padding_y * dpr as f32;
```

Comment:
Multiplying padding by dpr after computing lines/cols seems like a bug. Let's get rid of dpr from size info all together, assuming that everything user provided is dpr scaled already?

Reply:
> Let's get rid of dpr from size info all together


And how do you plan to do that?

Reply:
I'm trying to reload my DPI/DPR knowledge now... Is there anything inside `alacritty_terminal` at all that really needs DPR?

Reply:
From IRC, sounds like this could be the case, but should be considered in another PR so as to avoid too much more creep on this already very expanded issue.

Reply:
> I'm trying to reload my DPI/DPR knowledge now... Is there anything inside alacritty_terminal at all that really needs DPR?

No, there is nothing needing DPR in `alacritty_terminal`. In fact there's nothing in `alacritty_terminal` that needs padding or width/height either. The only thing `alacritty_terminal` needs is cols/lines and cell dimensions. Technically we use width/height in the tty, but that's kind of buggy-ish anyways.

It's all a bit questionable though, because there are escape sequences to query the width/height of the window. Alacritty just doesn't support them.